### PR TITLE
Updates for UFS Applications

### PR DESCRIPTION
**Pull request by uturuncoglu**
This PR aims to update UFSATM interface to work with UFS MR Weather and HAFS Applications

### CODE REVIEW

File: cime_config/buildnml [hunk 1]
```diff
@@ -159,13 +160,21 @@ def prep_input(case, datestamp, nmlgen_input, ic_file_prefix, prefix, input_type
     # TODO: make it xmlchnage variable
     new_o3forc = True
 
+    # Query date and time
```
Comment (jedwards4b) [inline cime_config/buildnml:172]: This code appears to be repeated at line 1129 - can we make it a subroutine?

File: cime_config/buildnml [hunk 2]
```diff
@@ -864,28 +883,20 @@ def _create_namelist_chgres(case, confdir, config, infile, nmlgen, nmlgen_def, n
...
-        task_count[atm_grid] *= 4
+    atm_grid = case.get_value("ATM_GRID").replace('r', '')
+    tasks_per_node = 6
+    task_count = {"C96"  : 2*tasks_per_node, # 2 nodes
+                  "C192" : 2*tasks_per_node, # 2 nodes
+                  "C384" : 4*tasks_per_node, # 4 nodes
+                  "C768" : 6*tasks_per_node} # 6 nodes
 
     if atm_grid in task_count.keys():
+        case.set_value("tasks_per_node", str(tasks_per_node), subgroup="case.chgres")
```
Comment (jedwards4b) [inline cime_config/buildnml:901]: We really should not do this in buildnml - it requires that case.setup be rerun.

**Review by jedwards4b — CHANGES_REQUESTED** [review]

File: cime_config/buildnml (hunk 1 above)
Comment (uturuncoglu) [reply cime_config/buildnml:172]: done

File: cime_config/buildnml (hunk 2 above)
Comment (uturuncoglu) [reply cime_config/buildnml:901]: I did not test it but if all resolution works for 6x6 combination, then we could define it in the workflow definition. it requires additional test with low resolution cases.

File: cime_config/buildnml (hunk 1 above)
Comment (jedwards4b) [reply cime_config/buildnml:172]: Thanks

Comment (uturuncoglu) [reply cime_config/buildnml:172]: It created additional problems in the other pieces of the buildnml but those are also solved with https://github.com/ESCOMP/FV3GFS_interface/pull/10/commits/0ae2d8e0e33d3bb32290fd429a153aef11719124

Comment (uturuncoglu) [reply cime_config/buildnml:172]: Currently, running full test suit with GNU compiler to create a baseline

File: cime_config/buildnml (hunk 2 above)
Comment (uturuncoglu) [reply cime_config/buildnml:901]: I am not sure i will have a chance to test this before next week and we will start work on S2S only. So, if it is okay for you, i'll update the app as it is after merging this one. I am also testing model on Stampede.

**Review by jedwards4b — APPROVED** [review]

**Pull request merged**